File: app/app/celery.py
"""
Celery config for app project.
"""
import os

from celery import Celery  # type: ignore

os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'app.settings')

app = Celery('app')
app.config_from_object('django.conf:settings', namespace='CELERY')
app.autodiscover_tasks()

app.conf.beat_schedule = {
    'check_unpaid_basket_every_7_days': {
        'task': 'shop.tasks.check_unpaid_basket',
        'schedule': 86400 * 7,  # seconds per days * days
    },
}

# For append tasks create 'tasks.py' file:
#
# from celery import shared_task
#
# @shared_task
# def task():
#     pass

File: graphics-next/src/app/celery.ts
Next.js היא סביבת פיתוח מבוססת JavaScript, ולא תומכת בקוד Python כמו שנכתב בקובץ המקורי. עם זאת, אם אתה מנסה להפוך את הפונקציונליות של הקובץ לקוד JavaScript, זה יכול להיראות כך:

אני מניח שאתה משתמש בספרייה cron לקביעת משימות מתוזמנות ב-Node.js:

./app/app/celery.js
-----
```javascript
/**
 * Celery config for app project.
 */
const cron = require('node-cron');

// Assuming you have a shop module with a check_unpaid_basket function
const shop = require('./shop');

// Schedule tasks to be run on the server.
cron.schedule('0 0 * * 0', function() {
  console.log('running a task every week');
  shop.check_unpaid_basket();
});
```
-----
בקוד הזה, אני משתמש בספריית `node-cron` לקביעת משימה שתרוץ מדי שבוע (באמצעות הביטוי cron '0 0 * * 0'). המשימה מבצעת את הפונקציה `check_unpaid_basket` מהמודול `shop`.

אני מניח שיש לך מודול `shop` עם פונקציה `check_unpaid_basket` שאתה רוצה להריץ. אם לא, תצטרך להחליף את הקוד הזה עם הפונקציה שאתה רוצה להריץ.

אנא שים לב שהקוד הזה הוא רק דוגמה, ואתה תצטרך להתאים אותו לצרכים שלך.